### ISSUE TITLE
Unexpected cursor position change while creating formula

### ISSUE THREAD

**Issue by OzelotVanilla** [typora/typora-issues#4642]
- [x] Searched existing issues to avoid creating duplicates.

- [x] Confirmed that it can be reproduced in built-in themes without customized css.

- [x] Searched http://support.typora.io/

----

**Describe the bug**
When typing some LATEX formula, the cursor jump to the first of the line.

**To Reproduce**
1. Type `$\cos(\` (the auto-pair is on, that means, type one `$` or `(` for two).
2. See the cursor jump to the first.

**Expected behavior**
The cursor should not move to the first.

**Sample Markdown File**
```markdown
## Draw a right triangle to simplify the given expressions ($x > 0$) (P68)

### 57. $\cos(\sin^{-1}(x))$
```

**Desktop (please complete the following information):**
 - OS: Windows 10 21H1, 19043.1165 (I think it is not the error about system version).

**Typora Version**
0.11.8 (beta)

**Comment by OzelotVanilla** [comment]
Additionally, there is a render error:

$\sin(\cos^{-1}(\dfrac{x}{2}))$ is rendered as 
![image](https://user-images.githubusercontent.com/59910575/131987772-d693aca0-50ea-432b-8667-3b0e1089e145.png)

**Comment by skarukas** [comment]
+1, I've been running into this same bug on macOS 10.15.7 with version 0.11.5. There is similar cursor-jumping behavior in this case:

`$\text{This first equation}$ causes the second equation to become "AbAbA" when clicking after the first "A" and typing "b" twice: $AAA$`

Can open another ticket if needed.
As a quick fix, rolling back to 0.10.11 seems to fix both issues.

**Comment by criphc** [comment]
same on macOS 11.6 with version 0.11.5, when typing ``(`` after ``$\ln`` or type anything after ``$\ln($``, the cursor jumps backwards before the first ``$``.

**Comment by abnerlee** [comment]
Seems fixed in 1.0

$\sin(\cos^{-1}(\dfrac{x}{2}))$ issue goes to #4992
